### PR TITLE
Refactored Costmaps using new probabilistic models and exact inference.

### PR DESCRIPTION
**Pull request by tomsch420**
Merge David Ps PR first. This depends on it.

### CODE REVIEW

File: src/pycram/resolver/probabilistic/probabilistic_action.py [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+        :return: The probabilistic circuit
+        """
+        centered_model = self.create_model_with_center()
+        outer_event = self.center_event().complement()
+        limiting_event = Event({self.relative_x: portion.open(-2, 2),
+                                self.relative_y: portion.open(-2, 2)})
+        event = outer_event & limiting_event
+        # go.Figure(event.plot()).show()
+        result, _ = centered_model.conditional(event)
+        print(result)
```
Comment (Tigul) [inline src/pycram/resolver/probabilistic/probabilistic_action.py:135]: Remove

File: src/pycram/resolver/probabilistic/probabilistic_action.py [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+        ActionDesignatorDescription.__init__(self)
+        ProbabilisticAction.__init__(self, policy)
+        self.object_designator = object_designator
+        self.arms = arms
+        self.grasps = grasps
+
+    def default_policy(self) -> ProbabilisticCircuit:
+        return GaussianCostmapModel().create_model()
+
+    def sample_to_action(self, sample: List) -> MoveAndPickUpPerformable:
```
Comment (Tigul) [inline src/pycram/resolver/probabilistic/probabilistic_action.py:181]: Documentation

File: test/test_database_resolver.py [hunk 3]
```diff
@@ -142,6 +133,7 @@ def test_object_at_different_location(self):
         new_milk.remove()
         kitchen.remove()
 
+    @unittest.skip
```
Comment (Tigul) [inline test/test_database_resolver.py:136]: Should this be skipped?

Comment (tomsch420) [reply test/test_database_resolver.py:136]: yes, the complex case is currently unstable. It is also of question if this should be a testcase in general.

File: src/pycram/orm/queries/queries.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import sqlalchemy.orm
+from sqlalchemy import Select, alias
+from ..task import TaskTreeNode
+from ..action_designator import PickUpAction
+from ..base import Position, RobotState, Pose
+from ..object_designator import Object
+
+
+class Query:
```
[the remark this answers is not in the document]
Comment (tomsch420) [reply src/pycram/orm/queries/queries.py:9]: Yes, we are currently thinking about how to write query views in general and this is the starting point.

File: src/pycram/resolver/probabilistic/probabilistic_action.py [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+import plotly.graph_objects as go
+
+
+class ProbabilisticAction:
+    """
+    Abstract class for probabilistic actions.
+    """
+
+    @dataclass
+    class Variables:
```
[the remark this answers is not in the document]
Comment (tomsch420) [reply src/pycram/resolver/probabilistic/probabilistic_action.py:32]: Yes, this is the documented inner class that will be implemented by the subclasses. Hence, the documentation needs to be here. Furthermore, this gives the programmer some structure on what to do.

**Pull request merged**